docs_app/content/tutorial/8-subjects.md

```diff
@@ -0,0 +1,304 @@
+}
+
+interface OP_Subject<T> {
+    /** 
+     * Register an observer to be notified by this subject, and return
+     * a Subscription that will remove the observer when you call `unsubscribe()` on it.
+     */
+    subscribe(observer: OP_Observer<T>): Subcription;
+    /** Notify all registered observers of a value */
+    notifyObservers(value: T): void;
+}
+```
+
+That only leaves us with that pesky `notifyObservers` method, which takes a value and returns `void`. WHich happens to be exactly what our `next` method did on `OP_Observer` (and RxJS [`Observer`](API)). So let's change `notifyObservers` to `next`. And, after we factor this out, we have this:
+
+```ts
+interface OP_Observer<T> {
+    /** Handle a value notification */
+    next(value: T): void;
+}
+
+interface OP_Subject<T> extends OP_Observer<T> {
+    /** 
+     * Register an observer to be notified by this subject, and return
+     * a Subscription that will remove the observer when you call `unsubscribe()` on it.
+     */
+    subscribe(observer: OP_Observer<T>): Subcription;
+}
+```
+
+By adding a few additional methods we have exactly the same thing as what we have in RxJS:
+
+```ts
+interface Observer<T> {
+    /** Handle a value notification */
+    next(value: T): void;
+    /** Handle an error notification */
+    error(err: any): void;
+    /** Handle completion */
+    complete(): void;
+}
+
+interface Observable<T> {
+    /** 
+     * Register an observer to be notified, and return
+     * a Subscription that will remove the observer when you call `unsubscribe()` on it.
+     */
+    subscribe(observer: Observer<T>): Subcription;
+}
+
+interface Subject<T> extends Observable<T>, Observer<T> {
+}
+```
+
+## Subjects: Multicasting - The Primary Use Case
+
+If the primary use case isn't apparent from the APIs described above, it can be best reiterated as follows: Subjects are a type where many observers can be notified of the same value simulatenously. The primary use case of a subject is [multicasting](GL).
+
+In terms of RxJS and observables, this might still not seem clear. But think about it, if a subject is an [`Observer`](API), then it can be used to subscribe to an [`Observable`](API) by being passed to [`subscribe`](API). And if, when you call `next` on the subject, it notifies all observers who have subscribed to it (thus "registering" themselves to be "notified", per the [observer pattern](#subjects-are-from-the-observer-pattern)), then a [`Subject`](API) can be used to [multicast](GL) any observable like so:
+
+
+### Example: Basic Multicast With A Subject
+
+```ts
+// Create a subject
+const subject = new Subject();
+
+// Subscribe our consumers to the subject, not the source
+subject.subscribe(observerA);
+subject.subscribe(observerB);
+subject.subscribe(observerC);
+
+// Connect the source, through a single subscription, to all consumers by
+// pushing its values through the subject.
+const connectionSubscription = source$.subscribe(subject);
+```
+
```

Comment:
Since this is listed as a primary use case, is this how a typical developer should be using subjects and observables?

I often see Angular developers take the Observable returned from an http call and re-emit its value through a Subject. This seemed unnecessary to me, but that technique appears to be the "primary use case" as per the above?